### DISCUSSION TITLE
How do I use a Babel plugin only during build time?

### DISCUSSION THREAD

**Discussion by paambaati** [vercel/next.js#20356] [Help] [answered]
I'd like to use the [`babel-plugin-transform-remove-console`](https://www.npmjs.com/package/babel-plugin-transform-remove-console) plugin to remove all console statements, but only during build time.

Currently, this is what my `.babelrc` file looks like —

```json
{
    "presets": [
        "next/babel"
    ],
    "env": {
        "production": {
            "plugins": [
                [
                    "transform-remove-console",
                    {
                        "exclude": [
                            "error",
                            "warn"
                        ]
                    }
                ]
            ]
        }
    }
}

```

It works, but during dev time as well (with `NODE_ENV=production`).

Why would I use `NODE_ENV=production` during local development? To test production-like stuff locally.

How do I ensure this works only during `next` runs but not `nest start` runs so I can continue using `console` statements during development?

**Comment by kelvinlouis** [comment, marked as answer]
If I understand your issue correctly you want see log statements while you use the development server (`npm run dev`). And for builds you want to strip them?

You can actually write js in your rc files: https://babeljs.io/docs/en/configuration#javascript-configuration-files

You could read the NODE_ENV and decide to push the plugin or not.

Hope this helps.
